### PR TITLE
Fix .env regex constants

### PR DESCRIPTION
**Pull request by PiotrFLEURY**
Set `.env` regex list as constants.

This pull request fixes the nit related in this comment: https://github.com/flutter/flutter/pull/128668#discussion_r1253377454

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -1390,12 +1410,12 @@ abstract class FlutterCommand extends Command<void> {
   ///
   /// Returns a record of key and value as strings.
   MapEntry<String, String> _parseProperty(String line) {
-    final RegExp blockRegExp = RegExp(r'^\s*([a-zA-Z_]+[a-zA-Z0-9_]*)\s*=\s*"""\s*(.*)$');
+    final RegExp blockRegExp = RegExp(DotEnvRegex.block);
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:1413]: I believe what Zach meant was lifting these invocations of hte `RegExp` constructor out of these methods and into static fields that are only allocated once. I believe the string literals are already constant.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:1413]: yes, they should be final. the point here is not about making them constant, but about avoiding a new allocation each time the methods are invoked.

Comment (PiotrFLEURY) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: I moved all the `RegExp` constructors. They are now build only one time.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 2]
```diff
@@ -34,6 +34,26 @@ import 'target_devices.dart';
...
+  static final RegExp doubleQuoteValue = RegExp(r'^"(.*)"\s*(\#\s*.*)?$');
+
+  // Dot env value wrapped in single quotes regex (eg FOO='bar')
+  static final RegExp quoteValue = RegExp(r"^'(.*)'\s*(\#\s*.*)?$");
+
+  // Dot env value wrapped in back quotes regex (eg FOO=`bar`)
+  static final RegExp backQuoteValue = RegExp(r'^`(.*)`\s*(\#\s*.*)?$');
+
+  // Dot env value without quotes regex (eg FOO=bar)
+  static final RegExp noQuoteValue = RegExp(r'^([^#\n\s]*)\s*(?:\s*#\s*(.*))?$');
```
Comment (andrewkolos) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:54]: It is not obvious what the difference between the `property` and `noQuoteValue` regexes are. I can guess that `property` refers to any kind of entry, while `noQuoteValue` refers specifically to entries where the value is not quoted. Is there a way to make the difference more obvious both with the name and the comment (which uses the same example)?

**Review by andrewkolos — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 2 above)
Comment (PiotrFLEURY) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:54]: I agree. Naming is not obvious enough. I will try to fix it today.

Comment (PiotrFLEURY) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: I pushed a new version.

Variables have been renamed to be more explicit.
I also added new comment lines to precise which string will be matched

**Review by andrewkolos — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 2 above)
Comment (andrewkolos) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: Thanks!

Comment (PiotrFLEURY) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: You're welcome

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by christopherfujino** [comment]
manually set the google testing to green, this change shouldn't affect google targets

**Pull request merged**